Restructure and comment the tau embedding method

winterchristian opened the pull request:
#### PR description:

This PR restructures and cleans the [tau embedding method](https://twiki.cern.ch/twiki/bin/viewauth/CMS/TauEmbeddingSamplesUL) ([`TauAnalysis/MCEmbeddingTools`](https://github.com/cms-sw/cmssw/tree/master/TauAnalysis/MCEmbeddingTools)) so that it is possible to produce RUN 3 tau embedding samples with CMS submission workflows.
One goal was to remove the need of `--customise` or `--customise_commands` in the tau embedding `cmsDriver.py` commands.
This was possible due to modifiers and the possibility to add python config fragments to some `cmsDriver.py` steps.
Only the `HLT` step required additional modification of the `cmsDriver.py` [config builder](https://github.com/cms-sw/cmssw/pull/48408/files#diff-cdb26c43130acb8faa9c94529dd55a55671d2621695d7c5a2a47aa20a62e62df).

The other goal of this PR is to have more understandable, better commented code and therefore sustainable code.

#### Tau embedding `cmsDriver.py` commands
With the changes introduced in this PR, tau embedding samples can be produced using the following `cmsDriver.py` commands.

<details>
<summary>Selection</summary>

```shell
cmsDriver.py \
--step RAW2DIGI,L1Reco,RECO,PAT,FILTER:TauAnalysis/MCEmbeddingTools/Selection_FILTER_cff.makePatMuonsZmumuSelection \
--processName SELECT \
--data \
--scenario pp \
--conditions auto:run3_data \
--era Run3_2024 \
--eventcontent TauEmbeddingSelection \
--datatier RAWRECO \
--filein "root://cmsdcache-kit-disk.gridka.de:1094//store/data/Run2024C/Muon0/RAW/v1/000/380/115/00000/00979445-916c-42e2-8038-428d7bd4f176.root" \
--fileout ...
```

</details>
<details>
<summary>LHE and Cleaning</summary>

```shell
cmsDriver.py \
--step USER:TauAnalysis/MCEmbeddingTools/LHE_USER_cff.embeddingLHEProducerTask,RAW2DIGI,RECO \
--processName LHEembeddingCLEAN \
--data \
--scenario pp \
--conditions auto:run3_data \
--era Run3_2024 \
--eventcontent TauEmbeddingCleaning \
--datatier RAWRECO \
--procModifiers tau_embedding_cleaning,tau_embedding_mu_to_mu \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>Simulation Gen</summary>

```shell
cmsDriver.py TauAnalysis/MCEmbeddingTools/python/Simulation_GEN_cfi.py \
--step GEN,SIM,DIGI,L1,DIGI2RAW \
--processName SIMembeddingpreHLT \
--mc \
--beamspot DBrealistic \
--geometry DB:Extended \
--era Run3_2024 \
--conditions auto:phase1_2024_realistic \
--eventcontent TauEmbeddingSimGen \
--datatier RAWSIM \
--procModifiers tau_embedding_sim,tau_embedding_mu_to_mu \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>Simulation HLT</summary>

```shell
cmsDriver.py \
--step HLT:Fake2+TauAnalysis/MCEmbeddingTools/Simulation_HLT_customiser_cff.embeddingHLTCustomiser \
--processName SIMembeddingHLT \
--mc \
--beamspot DBrealistic \
--geometry DB:Extended \
--era Run3_2024 \
--conditions auto:phase1_2024_realistic \
--eventcontent TauEmbeddingSimHLT \
--datatier RAWSIM \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>Simulation Reco</summary>

```shell
cmsDriver.py \
--step RAW2DIGI,L1Reco,RECO,RECOSIM \
--processName SIMembedding \
--mc \
--beamspot DBrealistic \
--geometry DB:Extended \
--era Run3_2024 \
--conditions auto:phase1_2024_realistic \
--eventcontent TauEmbeddingSimReco \
--datatier RAW-RECO-SIM \
--procModifiers tau_embedding_sim \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>Merging</summary>

```shell
cmsDriver.py \
--step USER:TauAnalysis/MCEmbeddingTools/Merging_USER_cff.merge_step,PAT \
--processName MERGE \
--data \
--scenario pp \
--conditions auto:run3_data \
--era Run3_2024 \
--eventcontent TauEmbeddingMergeMINIAOD \
--datatier USER \
--procModifiers tau_embedding_merging \
--inputCommands 'keep *_*_*_*' \
--filein ... \
--fileout ...
```

</details>
<details>
<summary>NanoAOD</summary>

```shell
cmsDriver.py \
--step NANO:@TauEmbedding \
--data \
--scenario pp \
--conditions auto:run3_data \
--era Run3_2024 \
--eventcontent TauEmbeddingNANOAOD \
--datatier NANOAODSIM \
--filein ... \
--fileout ...
```

</details>


#### PR validation:

I had to disable the tests, as I first have to get RAW samples from tape to create a small set of samples I can use. The test `.root` files right now are only useable for the old structure. 
I will deliver tests in the next pull request.

**UPDATE**
I reactivated the unit tests as I realized I can use the RAW samples from the release validation tests.
Both, the unit tests as the release validation tests worked locally.

Commit:
disable InitialRecoCorrection

Commit:
add tau embedding process modifiers

Commit:
first version without customize, but not HLT step

Commit:
rewrite tau embedding HLT step, by adding customise functionality in cmsDriver HLT step

Commit:
add config fragment for embedding NanoAOD step with special NanoAOD embedding table

Commit:
remove not needed python config fragments and switch to a consistent naming

Commit:
remove not needed functionalities, which were used for some tests in the past

Commit:
add comments to filter python config fragment

Commit:
add comments to tau embedding python config fragments

Commit:
move tau embedding header files to interface folder

Commit:
add some documentation to the tau embedding method

Commit:
format code

Commit:
change tests to new structure

kpedro88 commented:
@smuzaffar will RAW files used as input for unit tests in `TauAnalysis/MCEmbeddingTools/test` automatically get cached?

winterchristian commented on TauAnalysis/MCEmbeddingTools/docs/cmsDriver_cmds_DATA_RUN3.sh:
```diff
@@ -0,0 +1,167 @@
+	--era Run3_2024 \
+	--eventcontent RAWRECO \
+	--datatier RAWRECO \
+	--outputCommands 'drop *_*_*_SELECT','drop recoIsoDepositedmValueMap_muIsoDepositTk_*_*','drop recoIsoDepositedmValueMap_muIsoDepositTkDisplaced_*_*','drop *_ctppsProtons_*_*','drop *_ctppsLocalTrackLiteProducer_*_*','drop *_ctppsDiamondLocalTracks_*_*','drop *_ctppsDiamondRecHits_*_*','drop *_ctppsDiamondRawToDigi_*_*','drop *_ctppsPixelLocalTracks_*_*','drop *_ctppsPixelRecHits_*_*','drop *_ct ...
```
Ah nice, yes I will try to do that.

winterchristian replied:
Hi @kpedro88, 

I have tried to implement this in this commit: https://github.com/KIT-CMS/cmssw/commit/6f56ace6fd97da4eec20718abfcccc782e972845
Can you please take a look, then I will add it to the pull request and change the `cmsDriver` commands.

I also have two questions regarding this:
1. I only need to change the `--eventcontent` parameter of the `cmsDriver` commands and not the `--datatier` parameter, right? I don't really get the difference.
2. If I run the NanoAOD step with the new custom EventContents and run `edmDumpEventContent` on the produced NanoAOD `.root` file, I now get an output of collections. This was always empty before. Do you know why this happens? Also, my MiniAOD and NanoAOD `.root` files are now smaller. This is strange as the `edmDumpEventContent` gives still the same output for the MiniAOD file.

Thanks in advance.

winterchristian replied:
Thanks for the explanation.
And it was a good hint, looking at the differences of the config dumps I guess the difference is, that the `OutputModule` changed from `NanoAODOutputModule` to default `PoolOutputModule`.
This leads to a different TTree structure of the NanoAOD files. I could adjust this by adding "NANOAOD" in the event content string, so that [this line](https://github.com/cms-sw/cmssw/blob/1a09f40225355ae98666808a2b781c50f664d55b/Configuration/Applications/python/ConfigBuilder.py#L738) triggers the use of the `NanoAODOutputModule`.

Now everything looks the same and I will merge the branch into this branch.

Commit:
fix typos

Commit:
refactor MuMuForEmbeddingSelector to simplify the handling of values

Commit:
debug, as the wrong import order breaks the embedding cleaning config

cmsbuild commented:
Pull request #48408 was updated. @antoniovilela, @civanch, @cmsbuild, @davidlange6, @fabiocos, @kpedro88, @mandrenguyen, @mdhildreth, @rappoccio can you please check and sign again.

winterchristian commented on TauAnalysis/MCEmbeddingTools/plugins/MuMuForEmbeddingSelector.cc:
```diff
@@ -133,36 +142,57 @@ void MuMuForEmbeddingSelector::produce(edm::Event &iEvent, const edm::EventSetup
-  iEvent.put(std::make_unique<float>(chosenZCand_largest->mass()), "newMass");
-  iEvent.put(std::make_unique<float>(ZmumuCandidates.size()), "nPairCandidates");
+
+  iEvent.put(std::make_unique<float>(chosenZCand->daughter(0)->masterClone().castTo<pat::MuonRef>()->pt()),
```
resolved in https://github.com/cms-sw/cmssw/pull/48408/commits/693f5adac0478fe947f3f5df63e46e7289cfe899

cmsbuild commented:
-1

**Failed Tests**: RelVals-INPUT 
**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/46993/summary.html
**COMMIT**: 7e231845e270256ae16279447f1a9be8003c35b3
**CMSSW**: CMSSW_15_1_X_2025-06-29-0000/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/48408/46993/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals-INPUT

- [136.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/46993/runTheMatrixINPUT-results/136.902_RunDoubleMuon2016H)```136.902_RunDoubleMuon2016H/step2_RunDoubleMuon2016H.log```
- [136.903](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/46993/runTheMatrixINPUT-results/136.903_RunDoubleMuon2017B)```136.903_RunDoubleMuon2017B/step2_RunDoubleMuon2017B.log```
- [136.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/46993/runTheMatrixINPUT-results/136.901_RunDoubleMuon2016C)```136.901_RunDoubleMuon2016C/step2_RunDoubleMuon2016C.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.904](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/46993/runTheMatrixINPUT-results/136.904_RunDoubleMuon2018C)
</details>



## Comparison Summary



**Summary**:
 - You potentially [removed 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-06-29-0000+825b5c/69191/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 4 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-06-29-0000+825b5c/69191/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4065749
 - DQMHistoTests: [Total failures: 8161](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-06-29-0000+825b5c/69191/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4057568
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-06-29-0000+825b5c/69191/triggerResults): no differences found

winterchristian committed:
Add custom EventContent for tau embedding samples (#5)

* add special tau embedding EventContents

* add NANOAOD to NanoAOD event content name to be handled properly

* adapt eventcontent in cmsDriver commands

---------

Co-authored-by: Christian Winter <c.winter@kit.edu>

winterchristian commented on TauAnalysis/MCEmbeddingTools/test/run_2016postVFPUL_workflow_tests.sh:
```diff
@@ -12,102 +12,108 @@ function die {
-    --customise Configuration/DataProcessing/RecoTLR.customisePostEra_Run2_2016,TauAnalysis/MCEmbeddingTools/customisers.customiseLHEandCleaning_Reselect \
-    --filein file:selection.root \
+    --customise Configuration/DataProcessing/RecoTLR.customisePostEra_Run2_2016 \
+    --outputCommands 'drop *_*_*_SELECT','drop recoIsoDepositedmValueMap_muIsoDepositTk_*_*','drop recoIsoDepositedmValueMap_muIsoDepositTkDisplaced_*_*','drop *_ctppsProtons_*_*','drop *_ctppsLocalTrackLiteProducer_*_*','drop *_ctppsDiamondLocalTracks_*_*','drop *_ctppsDiamondRecHits_*_*','drop *_ctppsDiamondRawToDigi_*_*','drop *_ctppsPixelLocalTracks_*_*','drop *_ctppsPixelRecHits_*_*','drop * ...
```
resolved in 3f549be453b571b6e85889147043d41b3fafbbc6

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48408/45377

- Found files with invalid states:
  - TauAnalysis/MCEmbeddingTools/python/Merging_cff.py:
    - Added: 4276cf0f7b0e7c83c9ec2e369e9d5db0eb3b8466
    - Deleted: 0928b234df7c9cfb93d2d91449a24b9801222fe6

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/Applications/python/ConfigBuilder.py  modified in PR(s): #48162
  - File  Configuration/EventContent/python/EventContent_cff.py  modified in PR(s): #47859

cmsbuild commented:
Pull request #48408 was updated. @antoniovilela, @civanch, @cmsbuild, @davidlange6, @fabiocos, @kpedro88, @mandrenguyen, @mdhildreth, @rappoccio can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: UnitTests RelVals-INPUT 
**Size**: This PR adds an extra 48KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/47036/summary.html
**COMMIT**: 3f549be453b571b6e85889147043d41b3fafbbc6
**CMSSW**: CMSSW_15_1_X_2025-07-01-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/48408/47036/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test test-das-selected-lumis had ERRORS
</pre>

## RelVals-INPUT

- [136.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/47036/runTheMatrixINPUT-results/136.902_RunDoubleMuon2016H)```136.902_RunDoubleMuon2016H/step2_RunDoubleMuon2016H.log```
- [136.903](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/47036/runTheMatrixINPUT-results/136.903_RunDoubleMuon2017B)```136.903_RunDoubleMuon2017B/step2_RunDoubleMuon2017B.log```
- [136.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/47036/runTheMatrixINPUT-results/136.901_RunDoubleMuon2016C)```136.901_RunDoubleMuon2016C/step2_RunDoubleMuon2016C.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.904](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/47036/runTheMatrixINPUT-results/136.904_RunDoubleMuon2018C)
</details>



## Comparison Summary



**Summary**:
 - You potentially [removed 3 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-01-1100+825b5c/69221/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 4 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-01-1100+825b5c/69221/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4067195
 - DQMHistoTests: [Total failures: 8170](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-01-1100+825b5c/69221/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4059005
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-01-1100+825b5c/69221/triggerResults): found differences in 2 / 48 workflows

Commit:
fix spelling

Commit:
adapt cmsDriver commands for release validation and add tests for 2022

Commit:
switch to IBEos relval datasets in unittests and reactivate them

winterchristian commented on Configuration/Applications/python/ConfigBuilder.py:
```diff
@@ -1664,6 +1664,12 @@ def prepare_HLT(self, stepSpec = None):
+            # case where HLT:something was provided (most of the cases)
+            if '.' in stepSpec:
+                # case where HLT:path/to/config_cff.customiser.something was provided
+                # the customiser alowes to modify parts of the HLT menue
```
Thanks. Solved in https://github.com/cms-sw/cmssw/pull/48408/commits/ed3dfa771db3caefc7304aec653f0899f41cec91

cmsbuild commented:
Pull request #48408 was updated. @AdrianoDee, @Moanwar, @antoniovilela, @civanch, @cmsbuild, @davidlange6, @dickychant, @fabiocos, @kpedro88, @mandrenguyen, @mdhildreth, @miquork, @rappoccio, @srimanob, @subirsarkar can you please check and sign again.

winterchristian commented:
I forgot to adapt the release validation tests, sorry. I updated them in https://github.com/cms-sw/cmssw/pull/48408/commits/cee48c213641f93fa3622f3cb77d44e4bf50a57d, and they should now work (at least they work locally).
With https://github.com/cms-sw/cmssw/pull/48408/commits/2deb8beb5bfceaa80698d4cd28ebdc471f2e56ef I also reactivated the unit tests, as I realized I can use the RAW datasets from the release validation as input for the unit tests.

Please run the tests again.

cmsbuild commented:
+1

**Size**: This PR adds an extra 116KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/47068/summary.html
**COMMIT**: 2deb8beb5bfceaa80698d4cd28ebdc471f2e56ef
**CMSSW**: CMSSW_15_1_X_2025-07-03-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/48408/47068/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 3 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-03-1100+825b5c/69253/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 4 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-03-1100+825b5c/69253/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4067195
 - DQMHistoTests: [Total failures: 66](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-03-1100+825b5c/69253/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4067109
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-03-1100+825b5c/69253/triggerResults): no differences found

srimanob commented:
Hi @kpedro88 
RAW is cached in
`/eos/cms/store/user/cmsbuild/store/data/Run2022C/DoubleMuon/RAW/v1/000/356/381/00000/0b55fb47-4e17-49f2-a753-e625a741e44c.root`
Luminosities include 97-101, which is part of relvals definition,
https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_steps.py#L540

srimanob commented:
test parameters:
- workflows = 136.905,136.906
- relvals_opt = --what standard

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/47165/summary.html
**COMMIT**: 2deb8beb5bfceaa80698d4cd28ebdc471f2e56ef
**CMSSW**: CMSSW_15_1_X_2025-07-09-1800/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48408/47165/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-09-1800+825b5c/69327/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 6 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-09-1800+825b5c/69327/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4067201
 - DQMHistoTests: [Total failures: 26](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-09-1800+825b5c/69327/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4067155
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-09-1800+825b5c/69327/triggerResults): no differences found

srimanob commented:
Test of new workflows finish without crash.

With 136.905, there are errors show up in PAT step (step 7). We should take a look on this before start processing.

```
%MSG-e PFEGammaProducer:  PFEGammaProducer:particleFlowEGamma  09-Jul-2025 20:47:02 CEST Run: 356381 Event: 79811663
PFBLOCKALGO BUG!!!! Found a SuperCluster in a block by itself!
%MSG
%MSG-e PFEGammaAlgo::unwrapSuperCluster():  PFEGammaProducer:particleFlowEGamma  09-Jul-2025 20:47:02 CEST Run: 356381 Event: 79811663
There are no ECAL elements in a block with imported SC! This is a bug we should fix this!
%MSG
Begin processing the 4th record. Run 356381, Event 79889630, LumiSection 97 on stream 0 at 09-Jul-2025 20:47:04.064 CEST
Begin processing the 5th record. Run 356381, Event 79824642, LumiSection 97 on stream 0 at 09-Jul-2025 20:47:05.233 CEST
Begin processing the 6th record. Run 356381, Event 80370542, LumiSection 97 on stream 0 at 09-Jul-2025 20:47:06.079 CEST
Begin processing the 7th record. Run 356381, Event 80479704, LumiSection 97 on stream 0 at 09-Jul-2025 20:47:07.568 CEST
%MSG-e PFEGammaProducer:  PFEGammaProducer:particleFlowEGamma  09-Jul-2025 20:47:08 CEST Run: 356381 Event: 80479704
PFBLOCKALGO BUG!!!! Found a SuperCluster in a block by itself!
%MSG
%MSG-e PFEGammaAlgo::unwrapSuperCluster():  PFEGammaProducer:particleFlowEGamma  09-Jul-2025 20:47:08 CEST Run: 356381 Event: 80479704
There are no ECAL elements in a block with imported SC! This is a bug we should fix this!
%MSG
Begin processing the 8th record. Run 356381, Event 80391036, LumiSection 97 on stream 0 at 09-Jul-2025 20:47:09.120 CEST
Begin processing the 9th record. Run 356381, Event 79692970, LumiSection 97 on stream 0 at 09-Jul-2025 20:47:09.988 CEST
Begin processing the 10th record. Run 356381, Event 79921275, LumiSection 97 on stream 0 at 09-Jul-2025 20:47:11.009 CEST
%MSG-e PFEGammaProducer:  PFEGammaProducer:particleFlowEGamma  09-Jul-2025 20:47:11 CEST Run: 356381 Event: 79921275
PFBLOCKALGO BUG!!!! Found a SuperCluster in a block by itself!
%MSG
%MSG-e PFEGammaAlgo::unwrapSuperCluster():  PFEGammaProducer:particleFlowEGamma  09-Jul-2025 20:47:11 CEST Run: 356381 Event: 79921275
There are no ECAL elements in a block with imported SC! This is a bug we should fix this!
%MSG
```

missirol commented:
> One goal was to remove the need of --customise or --customise_commands in the tau embedding cmsDriver.py commands.

Just for my understanding: what is the downside of having to use `--customise` ? Is that not supported by some workflows ?

> Only the HLT step required additional modification of the cmsDriver.py [config builder](https://github.com/cms-sw/cmssw/pull/48408/files#diff-cdb26c43130acb8faa9c94529dd55a55671d2621695d7c5a2a47aa20a62e62df).

Fwiw, maybe a syntax like
```
-s HLT:menu+customisation1+customisation2
```
would be more intuitive than the
```
-s HLT:customisation.menu
```
one introduced in `prepare_HLT` by this PR (though, to me, the most intuitive thing would be to use `--customise`).

Martin-Grunewald commented:
How robust is the HLT modification against changes of the underlying HLT menu itself? It can not be that changes to HLT by TSG, breaking this tau embedding (and crashing it) would prevent HLT PRs being merged. For example, in 16_0 we'll change the Grun menu or 2025 menu to point to a Fake menu, where there is nothing for tau to embed, but the workflows are not allowed to crash!

winterchristian commented on TauAnalysis/MCEmbeddingTools/python/Simulation_HLT_customiser_cff.py:
```diff
@@ -0,0 +1,47 @@
+    process.hltPixelVertices = cms.EDProducer('EmbeddingHltPixelVerticesProducer')
+    process.offlinePrimaryVertices = cms.EDProducer('EmbeddingHltPixelVerticesProducer')
+    process.firstStepPrimaryVerticesUnsorted = cms.EDProducer('EmbeddingHltPixelVerticesProducer')
+    process.firstStepPrimaryVerticesPreSplitting = cms.EDProducer('EmbeddingHltPixelVerticesProducer')
```
This is something I just took over from the previous implementation: https://github.com/cms-sw/cmssw/blob/e152e7e2b26dd10cb2a20a5f1bc26ac6f8e4f0e6/TauAnalysis/MCEmbeddingTools/python/customisers.py#L646
I don't really know what lead to this line of code, but if you say this is not necessary, I will remove it. 
Can you confirm if the others are needed or if we are missing some?

winterchristian replied:
Ok looked into it and realized that it's a bug which I just copied from the old code.
I moved the modifications to the RECO step and tested it in https://github.com/cms-sw/cmssw/pull/48408/commits/f2d0b06c83d9f1a80c42cb556756532990c9d42e.

winterchristian commented:
> Just for my understanding: what is the downside of having to use `--customise` ? Is that not supported by some workflows ?

On my way to make this ready to run within CMS workflows, the feedback I got from O&C was to avoid customize. I don't know the reason behind it though. Maybe @srimanob can answer this.


> Fwiw, maybe a syntax like
> 
> ```
> -s HLT:menu+customisation1+customisation2
> ```
> 
> would be more intuitive

I agree and will implement this.

winterchristian commented:
> How robust is the HLT modification against changes of the underlying HLT menu itself? It can not be that changes to HLT by TSG, breaking this tau embedding (and crashing it) would prevent HLT PRs being merged. For example, in 16_0 we'll change the 2025 menu to point to a Fake menu, where there is nothing for tau to embed, but the workflows are not allowed to crash!

But is that realistic? These HLT changes only change the beamspot and the vertices. And even if their names change, these changes don't break anything. They would simply not overwrite existing modules, but create new ones with the old name. This means that in such a case, only the tau embedding method would work as if no changes had been made at all, which we would realize at the latest when validating the generated samples.
So I see no possibility how this would prevent HLT PRs being merged.

Commit:
change to more reasonable HLT customisation with '+' separator

Commit:
move the replacement of the PV from the HLT to the RECO step

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48408/45582

- Found files with invalid states:
  - TauAnalysis/MCEmbeddingTools/python/Merging_cff.py:
    - Added: 4276cf0f7b0e7c83c9ec2e369e9d5db0eb3b8466
    - Deleted: 0928b234df7c9cfb93d2d91449a24b9801222fe6

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_standard.py  modified in PR(s): #48490, #48606
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #48490, #48606

cmsbuild commented:
Pull request #48408 was updated. @AdrianoDee, @Moanwar, @antoniovilela, @civanch, @cmsbuild, @davidlange6, @dickychant, @fabiocos, @kpedro88, @mandrenguyen, @mdhildreth, @miquork, @rappoccio, @srimanob, @subirsarkar can you please check and sign again.

winterchristian commented on TauAnalysis/MCEmbeddingTools/python/Simulation_HLT_customiser_cff.py:
```diff
@@ -0,0 +1,47 @@
+    process.hltPixelVertices  = cms.EDProducer('EmbeddingHltPixelVerticesProducer')
+    # Replace HLT vertexing with vertex taken from LHE step
+    process.embeddingHltPixelVertices = cms.EDProducer('EmbeddingHltPixelVerticesProducer')
+    process.hltPixelVertices = cms.EDProducer('EmbeddingHltPixelVerticesProducer')
```
resolved in https://github.com/cms-sw/cmssw/pull/48408/commits/f2d0b06c83d9f1a80c42cb556756532990c9d42e

kpedro88 commented:
please test

civanch commented:
+1

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_16_0_X. Please open a backport if it should also go in to CMSSW_15_1_X.

winterchristian committed:
New Tau Embedding process modifiers  (#6)

* first try distributing process Modifiers of Cleaning step

* disable finale state radiation for mu->mu embedding

* update center of mass energy for run3

* implement correct ModifierChain for tau embedding process modifiers

* revert some changes

* set all process modifier for tau embeddign cleaning step

* Introduce special cleaning and simulation process modifier

* add process modifiers for sim gen step

* Set process modifiers for reco sim step

* add process modifier for tau embedding merging step
Those "toReplaceWith" modifiers need to be applied as late as possible e.g. in StandardSequences.Reconstruction_Data_cff.
This is to avoid collisions with other modules that are cloned and modified versions from the original modules or "toModify" modifier calls that try to modify the original modules. They throw an exeption if they want to do this on the replaced merger modules.

* add autoNANO option for the tau embedding nanoaod files

* adjust test cmsDriver commands to work with the new process modifiers

---------

Co-authored-by: Christian Winter <c.winter@kit.edu>

Commit:
update the cmsDriver commands of the release validation

winterchristian commented:
Sorry I forgot to update the cmsDriver commands in the release validation. Should work now, at least it worked locally. :sweat_smile:

Moanwar commented:
@cmsbuild please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 76KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/48103/summary.html
**COMMIT**: 2993c3e252fafeb287e2003a1ab6bbbbcede08e4
**CMSSW**: CMSSW_16_0_X_2025-09-14-0000/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48408/48103/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-09-14-0000+825b5c/70277/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-09-14-0000+825b5c/70277/validateJR.html)
 - DQMHistoTests: Total files compared: 52
 - DQMHistoTests: Total histograms compared: 4345076
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-09-14-0000+825b5c/70277/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4345053
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 51 files compared)
 - Checked 223 log files, 194 edm output root files, 52 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-09-14-0000+825b5c/70277/triggerResults): no differences found

jfernan2 commented:
@winterchristian is there any wf to test part or all of the Tau Embedding chain?

winterchristian commented:
I just saw I had a typo in my comment, instead of relval 136.901-136.906 I wrote 36.901-36.906. Sorry for that, it is a stupid copy and paste mistake. Could this be the reason for the failed CI?

But as far as I understand the logs:
```
136.901_RunDoubleMuon2016C Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED Step6-PASSED Step7-PASSED  - time date Mon Sep 15 16:17:53 2025-date Mon Sep 15 16:17:52 2025; exit: 0 0 0 0 0 0 0 0
136.902_RunDoubleMuon2016H Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED Step6-PASSED Step7-PASSED  - time date Mon Sep 15 16:17:53 2025-date Mon Sep 15 16:17:53 2025; exit: 0 0 0 0 0 0 0 0
136.903_RunDoubleMuon2017B Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED Step6-PASSED Step7-PASSED  - time date Mon Sep 15 16:17:54 2025-date Mon Sep 15 16:17:53 2025; exit: 0 0 0 0 0 0 0 0
136.904_RunDoubleMuon2018C Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED Step6-PASSED Step7-PASSED  - time date Mon Sep 15 16:17:55 2025-date Mon Sep 15 16:17:54 2025; exit: 0 0 0 0 0 0 0 0
136.905_RunDoubleMuon2022C Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED Step6-PASSED Step7-PASSED  - time date Mon Sep 15 16:17:55 2025-date Mon Sep 15 16:17:54 2025; exit: 0 0 0 0 0 0 0 0
136.906_RunMuon2022E Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED Step6-PASSED Step7-PASSED  - time date Mon Sep 15 16:17:56 2025-date Mon Sep 15 16:17:55 2025; exit: 0 0 0 0 0 0 0 0
```
They all passed.

cmsbuild commented:
+1

**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-825b5c/48136/summary.html
**COMMIT**: 2993c3e252fafeb287e2003a1ab6bbbbcede08e4
**CMSSW**: CMSSW_16_0_X_2025-09-15-1100/el8_amd64_gcc12
**Additional Tests**: NANO
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48408/48136/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 4 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-09-15-1100+825b5c/70302/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-09-15-1100+825b5c/70302/validateJR.html)
 - DQMHistoTests: Total files compared: 52
 - DQMHistoTests: Total histograms compared: 4345076
 - DQMHistoTests: [Total failures: 70](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-09-15-1100+825b5c/70302/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4344986
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 51 files compared)
 - Checked 223 log files, 194 edm output root files, 52 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-09-15-1100+825b5c/70302/triggerResults): found differences in 1 / 50 workflows

**Nano size comparison Summary**:



| workflow | id | kb/ev | ref kb/ev | diff kb/ev | ev/s/thd | ref ev/s/thd | diff rate | mem/thd | ref mem/thd |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| NANOmc106Xul16v2 | 2500.0001 | 3.085 | 3.085 | 0.000 ( +0.0% ) | 3.69 | 3.61 | +2.3% | 2.806 | 2.777 |
| NANOmc106Xul17v2 | 2500.0002 | 3.144 | 3.144 | 0.000 ( +0.0% ) | 3.48 | 3.36 | +3.5% | 3.292 | 3.233 |
| NANOmc106Xul18v2 | 2500.0003 | 3.097 | 3.097 | 0.000 ( +0.0% ) | 3.54 | 3.41 | +3.8% | 3.253 | 3.204 |
| NANOdata106Xul16v2 | 2500.0101 | 1.539 | 1.539 | 0.000 ( +0.0% ) | 6.18 | 6.33 | -2.4% | 2.849 | 2.766 |
| NANOdata106Xul17v2 | 2500.0102 | 1.925 | 1.925 | 0.000 ( +0.0% ) | 4.14 | 4.02 | +3.0% | 3.022 | 2.919 |
| NANOdata106Xul18v2 | 2500.0103 | 1.890 | 1.890 | 0.000 ( +0.0% ) | 4.93 | 4.82 | +2.4% | 2.981 | 2.876 |
| NANOmcUL16APVreMINI | 2500.0201 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.70 | 1.67 | +1.8% | 2.998 | 2.958 |
| NANOmcUL16reMINI | 2500.0202 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.64 | 1.62 | +1.5% | 2.997 | 2.951 |
| NANOmcUL17reMINI | 2500.0203 | 0.022 | 0.022 | -0.000 ( -0.2% ) | 1.51 | 1.53 | -1.5% | 2.706 | 2.666 |
| NANOmcUL18reMINI | 2500.0204 | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.29 | 1.28 | +0.2% | 3.137 | 2.943 |
| NANOdataUL16APVreMINI | 2500.0301 | 0.035 | 0.035 | 0.000 ( +0.0% ) | 1.28 | 1.27 | +0.8% | 2.909 | 2.885 |
| NANOdataUL16reMINI | 2500.0302 | 0.036 | 0.036 | 0.000 ( +0.1% ) | 1.49 | 1.47 | +1.1% | 2.946 | 2.894 |
| NANOdataUL17reMINI | 2500.0303 | 0.037 | 0.037 | 0.000 ( +0.0% ) | 1.26 | 1.26 | +0.2% | 2.939 | 2.896 |
| NANOdataUL18reMINI | 2500.0304 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.28 | 1.29 | -1.3% | 2.939 | 2.896 |
| NANOmc130X | 2500.1001 | 3.220 | 3.220 | 0.000 ( +0.0% ) | 4.26 | 4.17 | +2.1% | 3.242 | 3.232 |
| NANOdata130Xrun3 | 2500.1101 | 1.799 | 1.799 | 0.000 ( +0.0% ) | 6.27 | 6.47 | -3.0% | 2.828 | 2.814 |
| ScoutingNANOdata130Xrun3 | 2500.1301 | 0.771 | 0.771 | 0.000 ( +0.0% ) | 33.04 | 33.84 | -2.3% | 1.688 | 1.691 |
| muPOGNANOmc140X | 2500.2201 | 2.065 | 2.065 | 0.000 ( +0.0% ) | 5.38 | 5.61 | -4.0% | 2.369 | 2.353 |
| EGMNANOmc140X | 2500.2202 | 3.876 | 3.876 | 0.000 ( +0.0% ) | 3.88 | 4.07 | -4.6% | 2.727 | 2.728 |
| BTVNANOmc140X | 2500.2203 | 10.607 | 10.607 | 0.000 ( +0.0% ) | 2.50 | 2.60 | -3.8% | 2.949 | 2.930 |
| jmeNANOmc140X | 2500.2204 | 6.666 | 6.666 | 0.000 ( +0.0% ) | 1.12 | 1.17 | -4.7% | 2.801 | 2.799 |
| lepTrackInfoNANOmc140X | 2500.2206 | 3.570 | 3.570 | 0.000 ( +0.0% ) | 3.90 | 4.07 | -4.3% | 2.718 | 2.703 |
| ScoutingNANOmc140X | 2500.2207 | 1.486 | 1.486 | 0.000 ( +0.0% ) | 15.26 | 15.60 | -2.2% | 2.036 | 2.042 |
| ScoutingNANOwithPromptmc140X | 2500.2208 | 4.474 | 4.474 | 0.000 ( +0.0% ) | 3.34 | 3.21 | +4.0% | 2.967 | 2.948 |
| muPOGNANO140Xrun3 | 2500.2301 | 1.469 | 1.469 | 0.000 ( +0.0% ) | 7.57 | 7.80 | -2.8% | 2.670 | 2.517 |
| EGMNANOdata140Xrun3 | 2500.2302 | 2.623 | 2.623 | 0.000 ( +0.0% ) | 5.56 | 5.78 | -3.8% | 3.029 | 3.018 |
| BTVNANOdata140Xrun3 | 2500.2303 | 5.612 | 5.612 | 0.000 ( +0.0% ) | 3.57 | 3.76 | -5.0% | 3.241 | 3.224 |
| jmeNANOdata140Xrun3 | 2500.2304 | 3.976 | 3.976 | 0.000 ( +0.0% ) | 1.45 | 1.49 | -2.9% | 3.065 | 2.810 |
| lepTrackInfoNANOdata140Xrun3 | 2500.2306 | 2.413 | 2.413 | 0.000 ( +0.0% ) | 5.60 | 5.83 | -4.0% | 3.017 | 2.992 |
| ScoutingNANOdata140Xrun3 | 2500.2307 | 1.101 | 1.101 | 0.000 ( +0.0% ) | 28.65 | 30.02 | -4.6% | 1.679 | 1.683 |
| ScoutingNANOwithPromptdata140Xrun3 | 2500.2308 | 2.783 | 2.783 | 0.000 ( +0.0% ) | 5.04 | 4.83 | +4.5% | 3.237 | 3.148 |
| L1ScoutingNANOdata140Xrun3 | 2500.2309 | 0.082 | 0.082 | 0.000 ( +0.0% ) | 3.85 | 3.95 | -2.4% | 0.995 | 0.999 |
| L1ScoutingSelectionNANOdata140Xrun3 | 2500.231 | 0.081 | 0.081 | 0.000 ( +0.0% ) | 40.24 | 40.83 | -1.4% | 0.858 | 0.870 |
| l1DPGNANO140Xrun3 | 2500.2401 | 9.404 | 9.404 | 0.000 ( +0.0% ) | 5.19 | 7.45 | -30.3% | 1.939 | 1.941 |
| muDPGNANO140Xrun3 | 2500.2402 | 10.377 | 10.377 | 0.000 ( +0.0% ) | 1.64 | 1.52 | +8.2% | 1.750 | 1.739 |
| muDPGNANOBkg140Xrun3 | 2500.2403 | 6.401 | 6.401 | 0.000 ( +0.0% ) | 13.32 | 13.01 | +2.4% | 1.280 | 1.259 |
| hcalDPGNANO140Xrun3 | 2500.2404 | 486.016 | 486.016 | 0.000 ( +0.0% ) | 1.12 | 1.15 | -2.5% | 1.719 | 1.736 |
| hcalDPGCalibNANO140Xrun3 | 2500.2405 | 826.413 | 826.413 | 0.000 ( +0.0% ) | 1.50 | 1.55 | -2.8% | 1.729 | 1.704 |
| hcalDPGMCNANO140Xrun3 | 2500.2501 | 645.333 | 645.333 | 0.000 ( +0.0% ) | 1.65 | 1.67 | -0.9% | 1.817 | 1.813 |
| NANOmc2024reMINI | 2500.2601 | 0.021 | 0.021 | 0.000 ( +0.0% ) | 1.67 | 1.74 | -4.0% | 3.137 | 3.133 |
| NANOdata2024reMINI | 2500.2701 | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.46 | 1.49 | -2.1% | 2.883 | 2.879 |
| NANOmc150X | 2500.3001 | 3.217 | 3.217 | 0.000 ( +0.0% ) | 17.60 | 15.96 | +10.3% | 2.770 | 2.774 |
| NANOdata150X | 2500.3101 | 2.144 | 2.144 | 0.000 ( +0.0% ) | 33.32 | 35.46 | -6.0% | 2.497 | 2.500 |
| NANOdata150X | 2500.3102 | 2.630 | 2.630 | 0.000 ( +0.0% ) | 23.21 | 24.07 | -3.6% | 2.586 | 2.583 |
| muPOGNANOmc150X | 2500.3201 | 2.255 | 2.255 | 0.000 ( +0.0% ) | 15.36 | 15.34 | +0.1% | 2.658 | 2.650 |
| EGMNANOmc150X | 2500.3202 | 4.111 | 4.111 | 0.000 ( +0.0% ) | 14.01 | 13.36 | +4.9% | 2.756 | 2.752 |
| BTVNANOmc150X | 2500.3203 | 10.552 | 10.552 | 0.000 ( +0.0% ) | 4.76 | 3.71 | +28.4% | 2.972 | 2.965 |
| jmeNANOmc150X | 2500.3204 | 6.760 | 6.760 | 0.000 ( +0.0% ) | 1.34 | 1.16 | +15.4% | 2.940 | 2.324 |
| jmeNANOrePuppimc150X | 2500.3205 | 6.763 | 6.763 | 0.000 ( +0.0% ) | 1.26 | 1.12 | +12.2% | 3.207 | 2.578 |
| lepTrackInfoNANOmc150X | 2500.3206 | 3.721 | 3.721 | 0.000 ( +0.0% ) | 14.64 | 12.92 | +13.3% | 2.752 | 2.746 |
| ScoutingNANOmc150X | 2500.3207 | 1.562 | 1.562 | 0.000 ( +0.0% ) | 18.91 | 17.36 | +8.9% | 2.571 | 2.526 |
| ScoutingNANOwithPromptmc150X | 2500.3208 | 4.774 | 4.774 | 0.000 ( +0.0% ) | 10.53 | 9.81 | +7.4% | 2.996 | 2.341 |
| BPHNANOmc150X | 2500.3209 | 4.109 | 4.109 | 0.000 ( +0.0% ) | 16.98 | 15.44 | +9.9% | 2.829 | 2.157 |
| muPOGNANO150Xrun3 | 2500.3301 | 1.125 | 1.125 | 0.000 ( +0.0% ) | 24.79 | 21.73 | +14.1% | 2.413 | 2.404 |
| EGMNANOdata150Xrun3 | 2500.3302 | 2.985 | 2.985 | 0.000 ( +0.0% ) | 22.05 | 19.53 | +12.9% | 2.496 | 2.489 |
| BTVNANOdata150Xrun3 | 2500.3303 | 7.971 | 7.971 | 0.000 ( +0.0% ) | 5.81 | 5.23 | +11.1% | 2.725 | 2.423 |
| jmeNANOdata150Xrun3 | 2500.3304 | 4.349 | 4.349 | 0.000 ( +0.0% ) | 1.65 | 1.48 | +11.4% | 2.648 | 2.371 |
| jmeNANOrePuppidata150Xrun3 | 2500.3305 | 4.332 | 4.332 | 0.000 ( +0.0% ) | 1.44 | 1.32 | +8.8% | 2.940 | 2.582 |
| lepTrackInfoNANOdata150Xrun3 | 2500.3306 | 2.625 | 2.625 | 0.000 ( +0.0% ) | 24.29 | 21.78 | +11.5% | 2.494 | 1.885 |
| ScoutingNANOdata150Xrun3 | 2500.3307 | 1.737 | 1.737 | 0.000 ( +0.0% ) | 30.54 | 27.77 | +9.9% | 1.701 | 1.529 |
| ScoutingNANOwithPromptdata150Xrun3 | 2500.3308 | 2.601 | 2.601 | 0.000 ( +0.0% ) | 17.40 | 16.20 | +7.4% | 2.818 | 2.773 |
| BPHNANOdata150Xrun3 | 2500.3309 | 2.604 | 2.604 | 0.000 ( +0.0% ) | 32.37 | 29.95 | +8.1% | 2.541 | 2.487 |
| TTbarMINIAOD10.6_UL18v2 | 2500.9001 | 1.819 | 1.819 | 0.000 ( +0.0% ) | 27.82 | 26.71 | +4.2% | 1.943 | 1.940 |
| TTbarMINIAOD14.0 | 2500.9002 | 1.665 | 1.665 | 0.000 ( +0.0% ) | 27.11 | 24.69 | +9.8% | 1.586 | 1.581 |
| DYToLL_M-50_13TeV_pythia8 | 2500.9101 | 14.345 | 14.345 | 0.000 ( +0.0% ) | 7.09 | 6.82 | +4.0% | 1.103 | 1.101 |
| DYToll01234Jets_5f_LO_MLM_Madgraph_LHE_13TeV | 2500.9102 | 0.199 | 0.000 | 0.199 ( +0.0% ) | 2.87 | 3.42 | -16.3% | 0.856 | 0.836 |
| TTbar_Pow_LHE_13TeV | 2500.9103 | 0.110 | 0.110 | 0.000 ( +0.0% ) | 2.62 | 2.46 | +6.2% | 0.861 | 0.856 |


## NANO Comparison Summary



**Summary**:
 - You potentially [added 523 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_16_0_X_2025-09-15-1100+825b5c/70301/validateJR/logRootQA-events.log) to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_16_0_X_2025-09-15-1100+825b5c/70301/validateJR.html)
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 97320
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_16_0_X_2025-09-15-1100+825b5c/70301/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 97320
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 21 files compared)
 - Checked 138 log files, 82 edm output root files, 22 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_16_0_X_2025-09-15-1100+825b5c/70301/triggerResults): no differences found

civanch commented:
+simulation
